addons/l10n_in_ewaybill_stock/models/l10n_in_ewaybill.py

```diff
@@ -0,0 +1,566 @@
+
+    # Ewaybill details generated from the API
+    name = fields.Char("e-Way bill Number", copy=False, readonly=True, tracking=True)
+    ewaybill_date = fields.Date("e-Way bill Date", copy=False, readonly=True, tracking=True)
+    ewaybill_expiry_date = fields.Date("e-Way bill Valid Upto", copy=False, readonly=True, tracking=True)
+
+    state = fields.Selection(string='Status', selection=[
+        ('pending', 'Pending'),
+        ('generated', 'Generated'),
+        ('cancel', 'Cancelled'),
+    ], required=True, readonly=True, copy=False, tracking=True, default='pending')
+
+    # Stock picking details
+    picking_id = fields.Many2one("stock.picking", "Stock Transfer", copy=False)
+    move_ids = fields.One2many(
+        comodel_name='stock.move',
+        related="picking_id.move_ids",
+        inverse_name='ewaybill_id',
+        store=True
+    )
+    picking_type_code = fields.Selection(related='picking_id.picking_type_id.code')
+
+    # Document details
+    document_date = fields.Datetime("Document Date", compute="_compute_document_details", store=True)
+    document_number = fields.Char("Document", compute="_compute_document_details", store=True)
+    company_id = fields.Many2one("res.company", compute="_compute_document_details", store=True)
+    company_currency_id = fields.Many2one(related="company_id.currency_id")
+    supply_type = fields.Selection(string="Supply Type", selection=[
+        ("O", "Outward"),
+        ("I", "Inward")
+    ], compute="_compute_supply_type")
+    partner_bill_from_id = fields.Many2one(
+        "res.partner",
+        string='Bill From',
+        compute="_compute_document_partners_details",
+        store=True,
+        readonly=False
+    )
+    partner_bill_to_id = fields.Many2one(
+        "res.partner",
+        string='Bill To',
+        compute="_compute_document_partners_details",
+        store=True,
+        readonly=False
+    )
+    partner_ship_from_id = fields.Many2one(
+        "res.partner",
+        string='Dispatch From',
+        compute="_compute_document_partners_details",
+        store=True,
+        readonly=False
+    )
+    partner_ship_to_id = fields.Many2one(
+        'res.partner',
+        string='Ship To',
+        compute='_compute_document_partners_details',
+        store=True,
+        readonly=False
+    )
+
+    # Fields to determine which partner details are editable
+    is_bill_to_editable = fields.Boolean(compute="_compute_is_editable")
+    is_bill_from_editable = fields.Boolean(compute="_compute_is_editable")
+    is_ship_to_editable = fields.Boolean(compute="_compute_is_editable")
+    is_ship_from_editable = fields.Boolean(compute="_compute_is_editable")
+
+    transaction_type = fields.Selection(
+        selection=[
+            ("inter_state", "Inter State"),
+            ("intra_state", "Intra State"),
+        ],
+        string="Transaction",
+        compute="_compute_transaction_type",
+        readonly=False,
```

Comment:
I m not sure why it is done like this? Let me reconfirm it with @jva-odoo

Reply:
As discussed with @jva-odoo, we want to give user the flexibility to edit the form since there are particular use-cases for that to it will remain as it is

May be you and @jva-odoo can have meeting schedule for it if dis-agreements still persists (Please let me know what you guys discuss and concludes :sob:)